File: docs/templates/file-structure/template.py
"""Mi Retiro Proyectado — nombre del módulo.

Propósito:
    Explicar la responsabilidad permanente del módulo.

Alcance:
    Indicar qué contrato cubre y qué queda fuera de este archivo.
"""

from __future__ import annotations


def funcion_de_ejemplo(valor: str) -> str:
    """Normaliza un valor de ejemplo sin aplicar lógica de negocio real."""
    # Validación defensiva: evita propagar valores vacíos cuando el contrato exige texto.
    if not valor:
        return "valor-pendiente"

    return valor.strip()

File: docs/templates/file-structure/template.js
"use strict";

/*
 * Mi Retiro Proyectado — Nombre del módulo JavaScript.
 *
 * Propósito: Describir la responsabilidad de interfaz, storage, eventos o API.
 * Alcance: Indicar qué modifica, qué conserva y qué no debe duplicar.
 */

(() => {
  const CLAVE_EJEMPLO = "miRetiroProyectado.ejemplo";

  function cargarEstadoEjemplo() {
    // El storage puede no existir en pruebas, navegación privada o navegadores restringidos.
    try {
      return window.sessionStorage.getItem(CLAVE_EJEMPLO);
    } catch (_error) {
      return null;
    }
  }

  cargarEstadoEjemplo();
})();
